### DISCUSSION TITLE
Question: How can I add buttons for <Pos1>, <End> and <go to the text cursor (caret)> to the menu?

### DISCUSSION THREAD

**Discussion by merlinuwe** [chrisgurney/obsidian-note-toolbar#56] [Q&A]
Thanks for your GREAT plugin.

I want to go to 

- the top or 
- bottom of the note and 
- to the current position of the text cursor (caret). 
 
How can I do so?

Idea: Is there a way to call the keys `<Pos1>`, `<End>` or is there an other solution?

**Comment by chrisgurney** [comment]
@merlinuwe Hmm, good question. I'm not as well-versed on the universe of plugins available. 

@FelipeRearden Maybe you or another person has a suggestion?

**Reply by chrisgurney** [reply]
Looks like there are commands available specifically on mobile (not seeing them on desktop) to `go to first line` and `go to last line` of the file, if that helps!

**Reply by FelipeRearden** [reply]
> the top or bottom of the note and

Editor Commands Remap plugin has the commands `go to start` and `go to end`

https://github.com/c4ctus5/editor-commands-remap

> @FelipeRearden Maybe you or another person has a suggestion?

@merlinuwe what you mean by **I want to go to the current position of the text cursor (caret).** ?

try the command `Focus on last Note` to see if is what you want.

**Comment by merlinuwe** [comment]
Thanks for all your answers.

I tried the Editor Commands Remap plugin. This plugins allows only to goStart and goEnd in _editing_ mode of obsidian. There is no command to jump to the text cursor (caret). (The caret is the last place, where the text cursor was positioned.  In this example you see the text cursor marked

![image](https://github.com/chrisgurney/obsidian-note-toolbar/assets/4138869/be9570f1-4a8e-48ee-ad3f-8c88dba0eb05)

after the letter "r". )

In _edit_ mode, goStart and goEnd work, but they also change the state (?) of the text cursor (caret) to the first/last line of the note. And it is not possible, to start writing immediately. First I have to click to the place, where writing should start.

In _view_ mode, goStart and goEnd do not work at all. A (manually set) option to make buttons inactive/invisible if they do not work in one of the views would be user-friendly. So one sees, which commands are available in the current view and which are not.

Suggestion: The simplest solution for the user would be to add at least a command to obsidian to jump to the last caret (which is not the first or last line). Typing should be possible without a preceding mouse-click.

**Comment by chrisgurney** [comment]
@merlinuwe The latest [beta release](https://github.com/chrisgurney/obsidian-note-toolbar/discussions/171) adds support for the execution of Dataview, Templater, and JavaScript (via JS Engine).

This might help you solve your use cases here, if you haven't already found something that works, that is!
